USFMToolsSharp/LinterModules/UnpairedEndMarkers.cs

```diff
@@ -0,0 +1,113 @@
+                {typeof(RQEndMarker),typeof(RQMarker)},
+                {typeof(SCEndMarker),typeof(SCMarker)},
+                {typeof(SUPEndMarker),typeof(SUPMarker)},
+                {typeof(TLEndMarker),typeof(TLMarker)},
+                {typeof(VAEndMarker),typeof(VAMarker)},
+                {typeof(VPEndMarker),typeof(VPMarker)},
+                {typeof(WEndMarker),typeof(WMarker)},
+                {typeof(XEndMarker), typeof(XMarker)},
+            };
+            foreach (Marker marker in input.Contents)
+            {
+                missingEndMarkers.AddRange(CheckChildMarkers(marker, input));
+            }
+            return missingEndMarkers;
+
+        }
+        /// <summary>
+        /// Iterates through all children markers
+        /// </summary>
+        /// <param name="input"></param>
+        /// <param name="root"></param>
+        /// <returns></returns>
+        public List<LinterResult> CheckChildMarkers(Marker input, USFMDocument root)
+        {
+            List<LinterResult> results = new List<LinterResult>();
+
+            foreach (Marker marker in input.Contents)
+            {
+                if (markerPairs.ContainsKey(marker.GetType()))
+                {
+                    results.AddRange(CheckOpenMarker(marker, root));
+                }
+                results.AddRange(CheckChildMarkers(marker, root));
+            }
+            return results;
+        }
+        /// <summary>
+        /// Checks Opening Marker for Unique End Marker 
+        /// </summary>
+        /// <param name="input"></param>
+        /// <param name="root"></param>
+        /// <returns></returns>
+        public List<LinterResult> CheckOpenMarker(Marker input, USFMDocument root)
+        {
+            List<int> markerPositions = new List<int>();
+            List<Marker> hierarchy = root.GetHierarchyToMarker(input);
+            List<Marker> siblingMarkers = new List<Marker>(hierarchy[hierarchy.Count - 2].Contents);
+            siblingMarkers.Reverse();
+            foreach (Marker sibling in siblingMarkers)
+            {
+                if (sibling.GetType() == input.GetType())
+                {
+                    markerPositions.Add(sibling.Position);
+                }
+                else if (sibling.GetType() == markerPairs[input.GetType()])
+                {
+                    if (markerPositions.Count > 0)
+                    {
+                        markerPositions.RemoveAt(markerPositions.Count - 1);
+                    }
+                }
+            }
+            List<LinterResult> results = new List<LinterResult>();
+            foreach (int loneMarkerPosition in markerPositions)
+            {
+                results.Add(new LinterResult
+                {
+                    Position = loneMarkerPosition,
+                    Level = LinterLevel.Error,
+                    Message = $"Missing Opening marker for {input.GetType().ToString().Split('.')[3]}"

```

Comment:
Same as the other one we want to use name.

Reply:
@rbnswartz Made changes